### PR TITLE
[RFR] Beers can be added to a floor

### PR DESCRIPTION
**Pull request by HeroicEric**
Uggly for sure

### CODE REVIEW

File: backend/test/controllers/floor_beer_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  alias WeWorkForBeer.FloorBeer
+  @valid_attrs %{beer_id: 1, floor_id: 9}
+  @invalid_attrs %{}
+
+  setup do
+    conn = conn() |> put_req_header("accept", "application/json")
+    {:ok, conn: conn}
+  end
+
+  test "creates and renders resource when data is valid", %{conn: conn} do
```
Comment (davekaro) [inline backend/test/controllers/floor_beer_controller_test.exs]: o_O tests in the file! Bad ass

File: backend/test/models/beer_test.exs [hunk 2]
```diff
@@ -37,4 +37,34 @@ defmodule WeWorkForBeer.BeerTest do
   test "find_by_uid/1 - returns nil if beer with uid does not exist" do
     assert nil == Beer.find_by_uid("123")
   end
+
+  test "search/2 - returns beers with names that match the search query" do
+    bear_beer = Repo.insert Beer.changeset(%Beer{}, %{uid: "123", name: "Grizzly Bear IPA"})
```
Comment (davekaro) [inline backend/test/models/beer_test.exs]: lol deprecated

Comment (davekaro) [inline backend/test/models/beer_test.exs]: `insert` -> `insert!`

File: backend/test/controllers/floor_beer_controller_test.exs (hunk 1 above)
Comment (HeroicEric) [inline backend/test/controllers/floor_beer_controller_test.exs]: lol what

File: backend/web/controllers/beer_controller.ex [hunk 3]
```diff
@@ -6,8 +7,17 @@ defmodule WeWorkForBeer.BeerController do
   plug :scrub_params, "beer" when action in [:create, :update]
   plug :action
 
+  def index(conn, %{"query" => query}) do
+    beers =
+      from(b in Beer, limit: 20)
+      |> Beer.search(query)
```
Comment (davekaro) [inline backend/web/controllers/beer_controller.ex]: looks like hieroglyphics

File: backend/web/controllers/beer_controller.ex [hunk 4]
```diff
@@ -6,8 +7,17 @@ defmodule WeWorkForBeer.BeerController do
...
+      |> Repo.all
+
+    render(conn, "index.json", beers: beers)
+  end
+
   def index(conn, _params) do
-    beers = Repo.all(Beer)
+    beers = from(b in Beer, limit: 20) |> Repo.all
```
Comment (davekaro) [inline backend/web/controllers/beer_controller.ex]: LINQ!

File: backend/test/models/beer_test.exs (hunk 2 above)
Comment (HeroicEric) [inline backend/test/models/beer_test.exs]: @davekaro > :dog:

File: frontend/app/components/floor-panel/component.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+export default Ember.Component.extend({
+  store: Ember.inject.service(),
+
+  isAddingBeer: false,
+
+  init() {
+    this._super(...arguments);
+
+    this.searchResults = Ember.A();
```
Comment (davekaro) [inline frontend/app/components/floor-panel/component.js]: `this.set('searchResults', Ember.A())` ??

can it be `set(this.searchResults, Ember.A())` or `set(this, 'searchResults', Ember.A())`

File: backend/test/controllers/floor_beer_controller_test.exs (hunk 1 above)
Comment (davekaro) [inline backend/test/controllers/floor_beer_controller_test.exs]: lawd I can't read nm

File: frontend/app/locations/template.hbs [hunk 6]
```diff
@@ -5,7 +5,7 @@
     </div>
 
     <div class="locations-list list-group">
-      {{#each model as |location|}}
+      {{#each model key="id" as |location|}}
```
Comment (davekaro) [inline frontend/app/locations/template.hbs]: useless - https://github.com/emberjs/ember.js/pull/11461

**Pull request merged**